src/Chapter2.hs

```diff
@@ -740,7 +747,8 @@ the list with only those lists that contain a passed element.
 
 🕯 HINT: Use the 'elem' function to check whether an element belongs to a list
 -}
-contains = error "contains: Not implemented!"
+contains :: a -> [[a]] -> [[a]]
+contains n = filter (elem n)
```

Comment:
I'm afraid, this don't compile. `elem` tests values on equality, and you can't compare values of any type without telling the compiler so. A proper fix would be add `Eq` constraint:

```suggestion
contains :: Eq a => a -> [[a]] -> [[a]]
contains n = filter (elem n)
```

but we explain typeclasses and constraints only in the next chapter. For now, it would be enough to just use some monomorphic type like `Int`:


```suggestion
contains :: Int -> [[Int]] -> [[Int]]
contains n = filter (elem n)
```

Reply:
I see. Thanks for the reply. I always taught that i can just let compiler to choose, but I overlooked them. Thanks again.